### DISCUSSION TITLE
MICCA surgtoolloc missing file(image640_blur)to run the preprocess ?

### DISCUSSION THREAD

**Discussion by lukigg** [Project-MONAI/tutorials#1437] [Q&A] [answered]
Hi,
I try to run this whole project I joined the challange but cant figure out whats is this file img_dir = "/raid/surg/image640_blur/"   it's used in notebook below and is needed for other notebooks that create files to run detection.

I would be very gratefull for any help.

https://github.com/Project-MONAI/tutorials/blob/main/competitions/MICCAI/surgtoolloc/preprocess_extract_images_from_video.ipynb

**Comment by KumoLiu** [comment]
Hi @lukigg, I think `img_dir` is the output dir which doesn't include in the challenge.

Hope it can help you, thanks!

**Reply by lukigg** [reply]
Thanks. thought it was something different as I run it and don't get anything inside 🤔. Do you know if this could be a case if I dont load on pictures on drive as they weight 108 GB and I only use a few for the run?
![image](https://github.com/Project-MONAI/tutorials/assets/57077003/8e6a8238-c70c-417b-8575-274146dda683)

**Reply by KumoLiu** [reply]
Could you please check the data format under `video_dir`?

**Reply by lukigg** [reply]
Yes sure, I checked and  it's str even thought the videos are mp4 so i probablt need to change the reading fiels command🤔
![image](https://github.com/Project-MONAI/tutorials/assets/57077003/167f7800-5ac8-4396-b448-2018cedbf22a)

**Reply by KumoLiu** [reply]
Hi @lukigg, could you please try this command?
```
from pathlib import Path
print(list(Path(video_dir).iterdir()))
```
And I think you are just checking the type of `video_dir` instead of the content under this folder.

**Reply by lukigg** [reply]
Yea my bad i get this output
[PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000002.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000003.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000001.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000004.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000006.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000005.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000000.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000011.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000032.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000012.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000018.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000021.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000026.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000019.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000022.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000015.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000033.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000016.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000029.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000009.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000030.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000013.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000014.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000024.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000027.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000008.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000025.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000028.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000010.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000023.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000017.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000020.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000031.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000007.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000036.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000035.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000037.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000034.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000038.mp4'), PosixPath('/content/drive/MyDrive/Colab Notebooks/Monai/video_clips/clip_000039.mp4')]

**Reply by KumoLiu** [reply, marked as answer]
Then you can also get the basename by using:
```
list(x.stem for x in Path(video_dir).iterdir())
```
Thanks!

**Reply by lukigg** [reply]
I figure out some mistake and now it works thank you so much for your help @KumoLiu .
